octodns/processor/filter.py

```diff
@@ -125,6 +127,35 @@ def _process(self, zone, *args, **kwargs):
     process_target_zone = _process
 
 
+class _NetworkValueBaseFilter(BaseProcessor):
+    def __init__(self, name, _list):
+        super().__init__(name)
+        self.networks = []
+        for value in _list:
+            try:
+                self.networks.append(ip_network(value))
+            except ValueError:
+                raise ValueError(f'{value} is not a valid CIDR to use')
+
+    def _process(self, zone, *args, **kwargs):
+        for record in zone.records:
+            if record._type not in ['A', 'AAAA']:
+                continue
+
+            if any(
+                ip_address(value) in network
+                for value, network in product(record.values, self.networks)
+            ):
```

Comment:
Nice. Like the use of `any` here and the `product` bit cleans things up nicely. Small tweak would avoid recreating the IP addresses for each network test.

```suggestion
            ips = [ip_address(value) for value in record.values]
            if any(
                ip in network
                for ip, network in product(ips, self.networks)
            ):
```

Reply:
thanks for this :)